src/aof.c

```diff
@@ -1578,16 +1578,17 @@ int loadAppendOnlyFiles(aofManifest *am) {
             serverAssert(ai->file_type == AOF_FILE_TYPE_INCR);
             aof_name = (char*)ai->file_name;
             updateLoadingFileName(aof_name);
-            ++aof_num;
+            last_file = ++aof_num == total_num;
             start = ustime();
             ret = loadSingleAppendOnlyFile(aof_name);
-            if (ret == AOF_OK || (ret == AOF_TRUNCATED && aof_num == total_num)) {
+            if (ret == AOF_OK || (ret == AOF_TRUNCATED && last_file)) {
                 serverLog(LL_NOTICE, "DB loaded from incr file %s: %.3f seconds", 
                     aof_name, (float)(ustime()-start)/1000000);
-                ret = AOF_OK;
             }
 
-            if (ret == AOF_NOT_EXIST || ret == AOF_TRUNCATED) ret = AOF_FAILED;
+            if (ret == AOF_NOT_EXIST || (ret == AOF_TRUNCATED && !last_file)) {
+                ret = AOF_FAILED;
+            }
 
             if (ret != AOF_OK && ret != AOF_EMPTY) {
```

Comment:
same here?

Reply:
Changed to `if (ret == AOF_OPEN_ERR || ret == AOF_FAILED) `